fairworkflows/rdf_wrapper.py

```diff
@@ -28,14 +29,38 @@ def is_modified(self) -> bool:
         """Returns true if the RDF has been modified since initialisation"""
         return self._is_modified
 
+    def remove_non_attribute_triples(self, keep_triples: List[Tuple] = None):
+        """
+        Remove triples from RDF that are not attributes, i.e. the subject is not referring to the
+        URI of this concept
+
+        Args:
+            keep_triples: list of triples to keep. NB they do not have to be bound: (None,
+                RDF.type, None) will keep triples with RDF.type predicate.
+        """
+        if keep_triples is None:
+            keep_triples = []
+
+        def _keep(triple):
+            for keep_triple in keep_triples:
+                # If all of the s, p, o match (None or equal) we should keep
+                if all(keep is None or keep == x
```

Comment:
I don't like implementing this myself, but I couldn't figure out how to do it with rdflib. Any idea?

Reply:
Hmm yeah, I'm not sure you can 'negate' patterns in rdflib so often you do need to write your own function...

Reply:
Check